shiny/BrownDog/server.R

```diff
@@ -0,0 +1,127 @@
+library(shiny)
+library(leaflet)
+library(RPostgreSQL)
+library(PEcAn.DB)
+library(PEcAn.visualization)
+
+
+# Define server logic
+server <- shinyServer(function(input, output, session) {
+  output$agreementUI <- renderUI({
+    if (is.null(input$type)){
+      return()
+    }
+    
+    # Depending on input$type, we'll generate a different lincense agreement
+    switch(input$type,
+           "AmeriFlux" = checkboxInput("agreement", HTML("I agree to <a href='http://ameriflux.lbl.gov/data/data-policy/'>AmeriFlux license</a>."), value = FALSE, width = NULL),
+           "NARR" = checkboxInput("agreement", HTML("I agree to <a href='http://www.esrl.noaa.gov/psd/data/gridded/data.narr.html'>NARR license</a>."), value = FALSE, width = NULL),
+           "FLUXNET" = checkboxInput("agreement", HTML("I agree to FLUXNET license."), value = FALSE, width = NULL)
+    )
+  })
+  
+  observeEvent(input$type, {
+    # get all sites name, lat and lon by sitegroups
+    bety <- betyConnect("../../web/config.php")
+    con <- bety$con
+    on.exit(db.close(con))
+    sites <- db.query(paste0("SELECT sitename, ST_X(ST_CENTROID(geometry)) AS lon, ST_Y(ST_CENTROID(geometry))
```

Comment:
we should in the future make sure all queries are using dplyr instead, that way we get more typesafe queries and have somebody else worry how to optimize the return of results.